ProcessMaker/Repositories/TokenRepository.php

```diff
@@ -165,23 +167,37 @@ public function persistActivityActivated(ActivityInterface $activity, TokenInter
      * Validate email configuration and Send Email
      *
      * @param ActivityInterface $activity
+     * @param String $email
      */
-    private function validateAndSendActionByEmail(ActivityInterface $activity)
+    private function validateAndSendActionByEmail(ActivityInterface $activity, String $email)
     {
-        $isActionsByEmail = $activity->getProperty('isActionsByEmail', false);
-        if ($isActionsByEmail) {
+        $isActionsByEmail = $activity->getProperty('isActionsByEmail', "false");
+        if ($isActionsByEmail === "true") {
             $configEmail = json_decode($activity->getProperty('configEmail', []));
             // Get the parameters to send the email
-            $emailserver = $configEmail->emailServer ?? 0;
-            $subject = $configEmail->subject ?? '';
-            $emailScreen = $configEmail->screenEmailRef ?? 0;
-            $emailScreenCompleted = $configEmail->screenCompleteRef ?? 0;
-
-            //TODO send Email
+            $groupName = $configEmail->emailServer ?? 'Email Default Settings';
+            $configEmail->subject = $configEmail->subject ?? '';
+            $configEmail->screenEmailRef = $configEmail->screenEmailRef ?? 0;
+            $configEmail->screenCompleteRef = $configEmail->screenCompleteRef ?? 0;
+            
+
+            //Send Email
+            $this->loadConfig($groupName);
+            Mail::send(new TaskActionByEmail($configEmail, $email));
         }
 
     }
 
+    /**
```

Comment:
Please udpate the descrption here adding better description, defining the parameters, define a better name instead of `$groupName`